From 4c068e0ba07007d62bd8aad515d1f47370b7193b Mon Sep 17 00:00:00 2001
From: Karsten Sperling <ksperling@apple.com>
Date: Fri, 10 Oct 2025 22:29:18 +1300
Subject: [PATCH] Add OTBR_EXTERNAL_MBEDTLS build setting

This works the same as (and propagates to OT_EXTERNAL_MBEDTLS).
---
 etc/cmake/options.cmake               | 7 +++++++
 src/utils/CMakeLists.txt              | 2 +-
 src/web/CMakeLists.txt                | 2 +-
 tests/gtest/CMakeLists.txt            | 4 ++--
 third_party/openthread/CMakeLists.txt | 1 +
 tools/CMakeLists.txt                  | 4 ++--
 6 files changed, 14 insertions(+), 6 deletions(-)

--- a/etc/cmake/options.cmake
+++ b/etc/cmake/options.cmake
@@ -39,6 +39,13 @@ elseif (OTBR_MDNS STREQUAL "avahi")
     message(FATAL_ERROR "OTBR_MDNS=avahi is no longer supported. Use OTBR_MDNS=openthread or OTBR_MDNS=mDNSResponder.")
 endif()
 
+set(OTBR_EXTERNAL_MBEDTLS "" CACHE STRING "Specify external mbedtls library")
+if(NOT OTBR_EXTERNAL_MBEDTLS)
+    set(OTBR_MBEDTLS mbedtls) # from third_party/openthread
+else()
+    set(OTBR_MBEDTLS ${OTBR_EXTERNAL_MBEDTLS})
+endif()
+
 option(OTBR_BORDER_AGENT "Enable Border Agent" ON)
 if (OTBR_BORDER_AGENT)
     target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_BORDER_AGENT=1)
--- a/src/utils/CMakeLists.txt
+++ b/src/utils/CMakeLists.txt
@@ -43,5 +43,5 @@ add_library(otbr-utils
 target_link_libraries(otbr-utils PUBLIC
     otbr-common
     otbr-posix
-    mbedtls
+    ${OTBR_MBEDTLS}
 )
--- a/src/web/CMakeLists.txt
+++ b/src/web/CMakeLists.txt
@@ -50,7 +50,7 @@ target_link_libraries(otbr-web PRIVATE
     otbr-utils
     openthread-ftd
     openthread-posix
-    mbedtls
+    ${OTBR_MBEDTLS}
     pthread
 )
 install(
--- a/tests/gtest/CMakeLists.txt
+++ b/tests/gtest/CMakeLists.txt
@@ -48,7 +48,7 @@ add_executable(otbr-gtest-unit
     test_task_runner.cpp
 )
 target_link_libraries(otbr-gtest-unit
-    mbedtls
+    ${OTBR_MBEDTLS}
     otbr-common
     otbr-host
     otbr-utils
@@ -190,7 +190,7 @@ target_include_directories(otbr-gtest-ho
         ${OPENTHREAD_PROJECT_DIRECTORY}/tests/gtest
 )
 target_link_libraries(otbr-gtest-host-api
-    mbedtls
+    ${OTBR_MBEDTLS}
     otbr-common
     otbr-utils
     otbr-posix
--- a/third_party/openthread/CMakeLists.txt
+++ b/third_party/openthread/CMakeLists.txt
@@ -50,6 +50,7 @@ set(OT_DNS_CLIENT_OVER_TCP OFF CACHE STR
 set(OT_DNS_UPSTREAM_QUERY ${OTBR_DNS_UPSTREAM_QUERY} CACHE STRING "enable sending DNS queries to upstream" FORCE)
 set(OT_ECDSA ON CACHE STRING "enable ECDSA" FORCE)
 set(OT_EXTERNAL_HEAP ON CACHE STRING "enable external heap" FORCE)
+set(OT_EXTERNAL_MBEDTLS ${OTBR_EXTERNAL_MBEDTLS} CACHE STRING "Specify external mbedtls library" FORCE)
 if (OTBR_NFTABLES OR OTBR_PF)
     # otbr-agent owns the OTBR ingress firewall in-process (nftables or pf),
     # including producing the ingress prefix sets from Thread network data, so
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -33,7 +33,7 @@ target_link_libraries(pskc PRIVATE
     otbr-config
     otbr-common
     otbr-utils
-    mbedtls
+    ${OTBR_MBEDTLS}
 )
 
 add_executable(steering-data
@@ -43,7 +43,7 @@ target_link_libraries(steering-data PRIV
     otbr-config
     otbr-common
     otbr-utils
-    mbedtls
+    ${OTBR_MBEDTLS}
 )
 
 if ($ENV{REFERENCE_DEVICE})
